lib/vif-plug-providers/representor/vif-plug-representor.c

```diff
@@ -442,96 +376,79 @@ devlink_port_add_function(struct dl_port *port_entry,
             "UNKNOWN");
         return;
     };
-    /* Failure to fill key from typed values means calculation of the max key
-     * length is wrong, i.e. a bug. */
-    ovs_assert(fill_devlink_ports_key_from_typed(
-                            keybuf, sizeof(keybuf),
-                            *host_pf_mac, function_number));
-    shash_add(&devlink_ports, keybuf, xstrdup(port_entry->netdev_name));
+
+    struct eth_addr fallback_mac;
+    if (port_entry->flavour == DEVLINK_PORT_FLAVOUR_PCI_PF
+            && eth_addr_is_zero(port_entry->function.eth_addr)) {
+        /* PF representor does not have host facing MAC address set.
+         *
+         * For kernel versions where the devlink-port infrastructure does
+         * not provide MAC address for PCI_PF flavoured ports, there exist
+         * a interim interface in sysfs which is relative to the name of a
+         * PHYSICAL port netdev name.
+         *
+         * Note that there is not really any association between PHYSICAL and
+         * PF representor ports from the devlink data structure point of view.
+         * But we have found them to correlate on the devices where this is
```

Comment:
Good point. We expect physical (uplink) port numbering and the respective PF representor PF numbering to be consistent from the point of view of the same network controller. If that's not the case, I'd say it's a controller firmware bug.